js/main.js

```diff
@@ -1,103 +1,152 @@
+  //       <div class="row">
+  //         <div class="column-full justify-between">
+  //           <h2>Coffee</h2>
+  //           <i class="fas fa-pencil-alt"></i>
+  //         </div>
+  //         <div class="column-full">
+  //           <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione dicta voluptates reprehenderit
+  //             alias
+  //             nobis atque aperiam! Nobis cumque amet molestiae dicta voluptates reprehenderit alias
+  //             nobis atque aperiam!</p>
+  //         </div>
+  //       </div>
+  //     </div>
+  //   </div>
+  // </li>
 
   var $entryContainer = document.createElement('li');
   var $row = document.createElement('div');
-
-  $row.setAttribute('class', 'row');
-
   var $columnImage = document.createElement('div');
-  $columnImage.setAttribute('class', 'column-half');
-
   var $userImage = document.createElement('img');
-  $userImage.setAttribute('src', entry.photoUrl);
-  $userImage.setAttribute('class', 'width-100');
-  $userImage.setAttribute('alt', 'user-image');
-
   var $columnContent = document.createElement('div');
-  $columnContent.setAttribute('class', 'column-half');
-
+  var $columnHead = document.createElement('div');
+  var $columnNote = document.createElement('div');
   var $heading = document.createElement('h2');
-  $heading.textContent = entry.title;
+  var $icon = document.createElement('i');
+  var $subRow = document.createElement('row');
   var $note = document.createElement('p');
-  $note.textContent = entry.note;
 
   $entryContainer.appendChild($row);
   $row.appendChild($columnImage);
   $columnImage.appendChild($userImage);
   $row.appendChild($columnContent);
-  $columnContent.appendChild($heading);
-  $columnContent.appendChild($note);
+  $columnContent.appendChild($subRow);
+  $subRow.appendChild($columnHead);
+  $columnHead.appendChild($heading);
+  $columnHead.appendChild($icon);
+  $subRow.appendChild($columnNote);
+  $columnNote.appendChild($note);
+
+  $row.setAttribute('class', 'row');
+  $columnImage.setAttribute('class', 'column-half');
+  $userImage.setAttribute('src', entry.photoUrl);
+  $userImage.setAttribute('class', 'width-100');
+  $userImage.setAttribute('alt', 'user-image');
+  $columnContent.setAttribute('class', 'column-half');
+  $columnHead.setAttribute('class', 'column-full justify-between');
+  $heading.textContent = entry.title;
+  $icon.setAttribute('class', 'fas fa-pencil-alt');
+  $icon.setAttribute('data-entry-id', entry.entryId);
+  $columnNote.setAttribute('class', 'column-full');
+  $note.textContent = entry.note;
 
   return $entryContainer;
 }
 
-var $ul = document.querySelector('ul');
-
 function handleDOMContentLoaded(event) {
   for (var i = 0; i < data.entries.length; i++) {
     var entriesList = entryTemplate(data.entries[i]);
     $ul.appendChild(entriesList);
   }
   switchView(data.view);
+  for (var x = 0; x < $li.length; x++) {
+    $li[x].setAttribute('data-entry-id', data.entries[x].entryId);
+  }
```

Comment:
You should be doing this in your `entryTemplate` function, it's an important part of the DOM tree for each element.